hubblestack/extmods/hubble_mods/ssl_certificate.py

```diff
@@ -0,0 +1,316 @@
+    ports on a system and provides a 'host, port' tuple (or a list of host, port tuples)
+    to this module, this module will connect to the host and port and fetch
+    certificate details if a certificate is attached on the port. As an example,
+    osquery needs to provide the value in the following format.
+    +-------------------------------+-----------+
+    | host_ip                       | host_port |
+    +-------------------------------+-----------+
+    | 127.0.0.1                     | 80        |
+    | 2001:db8:85a3::8a2e:370:7334  | 80        |
+    | 127.0.0.1                     | 443       |
+    | 2001:db8:85a3::8a2e:370:7334  | 443       |
+    +-------------------------------+-----------+
+
+Comparator compatible with this module: certificate
+
+Sample Output:
+{
+ "execution_time": 0.006193876266479492,
+ "ssl_cert_pem": "-----BEGIN CERTIFICATE-----
+                  abcdxyz
+                  -----END CERTIFICATE-----\n",
+ "ssl_cert_version": "2",
+ "ssl_end_time": "2021-10-19 09:27:57",
+ "ssl_has_expired": false,
+ "ssl_issuer_common_name": "DigiCert SHA2 Secure Server CA",
+ "ssl_serial_number": "1",
+ "ssl_signature_algorithm": "sha256WithRSAEncryption",
+ "ssl_src_host": "127.0.0.1",
+ "ssl_src_port": "9100",
+ "ssl_start_time": "2020-10-19 09:27:57",
+ "ssl_subject_alternative_names": ["DNS:"],
+ "ssl_subject_country": "US",
+ "ssl_subject_organisation": "Adobe Systems Incorporated",
+ "ssl_subject_organisation_unit": "IT"
+ }
+"""
+import logging
+import OpenSSL
+import ssl
+import time
+from datetime import datetime
+from socket import setdefaulttimeout
+
+import hubblestack.extmods.module_runner.runner_utils as runner_utils
+from hubblestack.utils.hubble_error import HubbleCheckValidationError
+
+log = logging.getLogger(__name__)
+
+
+def validate_params(block_id, block_dict, extra_args=None):
+    """
+    Validate all mandatory params required for this module
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': {'host_ip': '127.0.0.1',
+                                                'host_port': 443},
+                                    'status': True},
+                  'caller': 'Audit'}
+
+    Raises:
+        HubbleCheckValidationError: For any validation error
+    """
+    log.debug('Module: ssl_certificate Start validating params for check-id: {0}'.format(block_id))
+
+    error = {}
+    chain_args = extra_args.get('chaining_args')
+    endpoint_chained = runner_utils.get_chained_param(chain_args)
+    if endpoint_chained:
+        host_ip = endpoint_chained.get('host_ip')
+        host_port = endpoint_chained.get('host_port')
+    else:
+        host_ip = runner_utils.get_param_for_module(block_id, block_dict, 'host_ip')
+        host_port = runner_utils.get_param_for_module(block_id, block_dict, 'host_port')
```

Comment:
Default value of port is 443 in older module.

Reply:
It is not default in case of ssl_certificate module